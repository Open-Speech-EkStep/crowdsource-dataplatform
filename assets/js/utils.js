const { HOUR_IN_SECONDS, SIXTY } = require("./constants");

const ALL_LANGUAGES = [
    {value: "Assamese",id: "as", text: "অসমীয়া", hasLocaleText: true},
    {value: "Bengali", id: "bn", text: "বাংলা", hasLocaleText: true},
    {value: "English", id: "en", text: "English", hasLocaleText: true},
    {value: "Gujarati", id: "gu", text: "ગુજરાતી", hasLocaleText: true},
    {value: "Hindi", id: "hi", text: "हिंदी", hasLocaleText: true},
    {value: "Kannada", id: "kn", text: "ಕನ್ನಡ", hasLocaleText: true},
    {value: "Malayalam", id: "ml", text: "മലയാളം", hasLocaleText: true},
    {value: "Marathi", id: "mr", text: "मराठी", hasLocaleText: true},
    {value: "Odia", id: "or", text: "ଓଡିଆ", hasLocaleText: true},
    {value: "Punjabi", id: "pa", text: "ਪੰਜਾਬੀ", hasLocaleText: true},
    {value: "Tamil", id: "ta", text: "தமிழ்", hasLocaleText: true},
    {value: "Telugu", id: "te", text: "తెలుగు", hasLocaleText: true}];

function convertPXToVH(px) {
    return px * (100 / document.documentElement.clientHeight);
}

function setPageContentHeight() {
    const $footer = $('footer');
    const $nav = $('.navbar');
    const edgeHeightInPixel = $footer.outerHeight() + $nav.outerHeight()
    const contentHeightInVH = 100 - convertPXToVH(edgeHeightInPixel)
    $('#content-wrapper').css('min-height', contentHeightInVH + 'vh');
}

function toggleFooterPosition() {
    const $footer = $('footer');
    $footer.toggleClass('fixed-bottom')
    $footer.toggleClass('bottom')
}

function fetchLocationInfo() {
    //https://api.ipify.org/?format=json
    return fetch('https://www.cloudflare.com/cdn-cgi/trace').then(res => res.text()).then(ipAddressText => {
        const dataArray = ipAddressText.split('\n');
        let ipAddress = "";
        for (let ind in dataArray) {
            if (dataArray[ind].startsWith("ip=")) {
                ipAddress = dataArray[ind].replace('ip=', '');
                break;
            }
        }
        if (ipAddress.length !== 0) {
            return fetch(`/location-info?ip=${ipAddress}`);
        } else {
            return new Promise((resolve, reject) => {
                reject("Ip Address not available")
            })
        }
    });
}

const performAPIRequest = (url) => {
    return fetch(url).then((data) => {
        if(!data.ok) {
            throw Error(data.statusText || 'HTTP error');
        } else {
            return Promise.resolve(data.json());
        }
    });
}

const getLocaleString = function() {
    performAPIRequest('/get-locale-strings')
        .then((response) => {
            localStorage.setItem('localeString', JSON.stringify(response));
        });
}

const updateLocaleLanguagesDropdown = (language) => {
    const dropDown = $('#localisation_dropdown');
    const localeLang = ALL_LANGUAGES.find(ele => ele.value === language);
    if (language.toLowerCase() === "english" || localeLang.hasLocaleText === false) {
        dropDown.html('<a id="english" class="dropdown-item" href="/changeLocale/en">English</a>');
    } else {
        dropDown.html(`<a id="english" class="dropdown-item" href="/changeLocale/en">English</a>
        <a id=${localeLang.value} class="dropdown-item" href="/changeLocale/${localeLang.id}">${localeLang.text}</a>`);
    }
}

const calculateTime = function (totalSeconds, isSeconds = true) {
    const hours = Math.floor(totalSeconds / HOUR_IN_SECONDS);
    const remainingAfterHours = totalSeconds % HOUR_IN_SECONDS;
    const minutes = Math.floor(remainingAfterHours / SIXTY);
    const seconds = Math.round(remainingAfterHours % SIXTY);
    if (isSeconds) {
        return { hours, minutes, seconds };
    } else {
        return { hours, minutes };
    }
};

const formatTime = function (hours, minutes = 0, seconds = 0) {
    let result = "";
    if (hours > 0) {
        result += `${hours} hrs `;
    }
    if (minutes > 0) {
        result += `${minutes} min `;
    }
    if (hours === 0 && minutes === 0 && seconds > 0) {
        result += `${seconds} sec `;
    }
    return result.substr(0, result.length - 1);
};

module.exports = { setPageContentHeight, toggleFooterPosition, fetchLocationInfo, updateLocaleLanguagesDropdown ,calculateTime, formatTime, getLocaleString, performAPIRequest}
