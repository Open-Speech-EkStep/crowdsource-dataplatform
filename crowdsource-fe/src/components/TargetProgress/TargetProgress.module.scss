.root {
  position: relative;
}

.percentage {
  font-family: Rowdies, 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-size: 14px;
  line-height: normal;

  @include media-breakpoint-up(md) {
    font-size: 24px;
  }
}

.count {
  color: var(--text-orange);
  font-size: 28px;

  @include media-breakpoint-up(md) {
    font-size: 40px;
  }
}

.hours {
  font-family: Roboto, sans-serif;
  font-weight: bold;
  font-size: 14px;
  line-height: 20px;

  @include media-breakpoint-up(md) {
    font-size: 20px;
    line-height: 28px;
  }
}
