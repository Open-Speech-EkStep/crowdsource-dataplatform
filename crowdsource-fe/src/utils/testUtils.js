import { render } from '@testing-library/react';

const Wrapper = ({ children }) => children;

const customRender = (ui, options) => render(ui, { wrapper: Wrapper, ...options });

// re-export everything
export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { renderHook } from '@testing-library/react-hooks';

// override render method
export { customRender as render };
