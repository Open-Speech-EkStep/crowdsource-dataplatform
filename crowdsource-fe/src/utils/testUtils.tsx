/* eslint-disable jest/no-export */
import type { ReactNode, ReactElement, ComponentType } from 'react';

import type { RenderOptions, RenderResult } from '@testing-library/react';
import { render, act } from '@testing-library/react';
import { axe } from 'jest-axe';
import puppeteer from 'puppeteer';
import type { Browser } from 'puppeteer';

const Wrapper: ComponentType = ({ children }: { children?: ReactNode }) => children as ReactElement;

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Wrapper, ...options });

// re-export everything
export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { renderHook } from '@testing-library/react-hooks';

// override render method
export { customRender as render };

export const verifyAxeTest = ({ container }: { container: RenderResult['container'] }) => {
  it('should not fail an axe audit', async () => {
    let axeResult;

    await act(async () => {
      axeResult = await axe(container);
    });

    expect(axeResult).toHaveNoViolations();
  });
};

export const verifyVRTest = (title: string, url: string) => {
  describe(`${title} VR test`, () => {
    let browser: Browser;

    beforeAll(async () => {
      browser = await puppeteer.launch({
        headless: true,
        devtools: false,
        ignoreHTTPSErrors: true,
        args: ['--start-fullscreen'],
      });
    });

    it('match snapshot', async () => {
      const page = await browser.newPage();
      await page.setViewport({ width: 1920, height: 1080 });

      await page.goto(url, {
        waitUntil: 'networkidle2',
      });

      await page.addStyleTag({
        content: `
      *,
      *::after,
      *::before {
          transition-delay: 0s !important;
          transition-duration: 0s !important;
          animation-delay: -0.0001s !important;
          animation-duration: 0s !important;
          animation-play-state: paused !important;
          caret-color: transparent !important;
      }
      `,
      });

      const image = await page.screenshot({
        fullPage: true,
      });

      expect(image).toMatchImageSnapshot();
    });

    afterAll(async () => {
      await browser.close();
    });
  });
};
