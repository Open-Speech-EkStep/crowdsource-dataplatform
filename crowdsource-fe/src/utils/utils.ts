import { jsPDF } from 'jspdf';
import { i18n } from 'next-i18next';

import { ErrorStatusCode } from 'constants/errorStatusCode';
import { INITIATIVES_MAPPING } from 'constants/initiativeConstants';
import { KEYBOARD_ERROR, LANGUAGE_UNICODE, OTHER_LANGUAGE_UNICODE } from 'constants/Keyboard';
import type { InitiativeType } from 'types/InitiativeType';

import apiPaths from '../constants/apiPaths';

var platform = require('platform');

export const convertIntoHrsFormat = (data: any, isSeconds = true) => {
  const hours = Math.floor(data / 3600);
  const remainingAfterHours = data % 3600;
  const minutes = Math.floor(remainingAfterHours / 60);
  const seconds = Math.round(remainingAfterHours % 60);
  if (isSeconds) {
    return { hours, minutes, seconds };
  } else {
    return { hours, minutes };
  }
};

export const roundOffValue = (average: any, precision: number) => {
  const formattedAverage = average.toFixed(precision);
  return formattedAverage;
};

export const capitalizeFirstLetter = (strValue: string) => {
  return strValue?.charAt(0).toUpperCase() + strValue?.slice(1);
};

const translateText = (text: string) => i18n?.t(text);

export const formatTime = (hours: number, minutes = 0, seconds = 0) => {
  const hrStr = translateText('hours1');
  const minStr = translateText('minutes1');
  const secStr = translateText('seconds1');
  let result = '';
  if (hours > 0) {
    result += `${hours} ${hrStr} `;
  }
  if (minutes > 0) {
    result += `${minutes} ${minStr} `;
  }
  if (hours === 0 && minutes === 0 && seconds > 0) {
    result += `${seconds} ${secStr} `;
  }

  if (hours === 0 && minutes === 0 && seconds === 0) {
    result += `0 ${secStr} `;
  }

  return result.substr(0, result.length - 1);
};

export const convertTimeFormat = (value: any) => {
  const { hours, minutes, seconds } = convertIntoHrsFormat(Number(value) * 60 * 60);
  const data = formatTime(hours, minutes, seconds);
  return data;
};

const roundDuration = (value: number) => {
  return Math.round(value * 100) / 100;
};

export const getHoursValue = (value: any) => {
  return roundDuration(value);
};

export const getMinutesValue = (value: any) => {
  return roundDuration(value * 60);
};

export const getHoursText = (value: any) => {
  return `${getHoursValue(value)} ${i18n?.t('hours2')}`;
};

export const getMinutesText = (value: any) => {
  return `${getMinutesValue(value)} ${i18n?.t('minutes2')}`;
};

export const isSunoOrBoloInitiative = (value: InitiativeType) => {
  const boloOrSuno = ['asr', 'text'];
  return boloOrSuno.includes(value);
};

export const isBoloInitiative = (value: InitiativeType) => {
  const boloOrSuno = ['text'];
  return boloOrSuno.includes(value);
};

export const findInputError = (text: string, currentModule: string, language: string) => {
  if (!text.length) return;
  const newText = text.replace(/\s/g, ''); //read input value, and remove "space" by replace \s
  let error: any = KEYBOARD_ERROR.language;
  const isSunoModule = currentModule === INITIATIVES_MAPPING.suno;
  const specialSymbols = /[\u0021-\u002F\u003A-\u0040\u005B-\u0060\u007B-\u007E\u0964-\u0965]/;
  if (currentModule === INITIATIVES_MAPPING.suno && specialSymbols.test(newText) === true) {
    return KEYBOARD_ERROR.symbol;
  }
  Object.entries(isSunoModule ? LANGUAGE_UNICODE : OTHER_LANGUAGE_UNICODE).forEach(([key, value]) => {
    if (value.test(newText)) {
      //Check Unicode to see which one is true
      if (key.toLowerCase() === language.toLowerCase()) {
        error = KEYBOARD_ERROR.noError;
      }
    }
  });
  return error;
};

export const fetchLocationInfo = async () => {
  const result = await fetch('https://www.cloudflare.com/cdn-cgi/trace')
    .then(res => res.text())
    .then(async ipAddressText => {
      const dataArray = ipAddressText.split('\n');
      let ipAddress = '';
      for (let ind in dataArray) {
        if (dataArray[ind].startsWith('ip=')) {
          ipAddress = dataArray[ind].replace('ip=', '');
          break;
        }
      }
      if (ipAddress.length !== 0) {
        const data = await fetch(`${apiPaths.locationInfo}?ip=${ipAddress}`);
        return data && data.json();
      } else {
        return new Promise((resolve, reject) => {
          reject('Ip Address not available');
        });
      }
    });

  return result;
};

export const getDeviceInfo = () => {
  const os = platform.os;
  return os.family + ' ' + os.version;
};

export const getBrowserInfo = () => {
  return platform.name + ' ' + platform.version;
};

export const downloadBadge = (
  locale: string | undefined,
  initiative: string,
  source: string,
  badgeType: string,
  badgeId: string
) => {
  const pdf = new jsPDF();
  const img = new Image();
  img.src = `/images/${locale}/badges/${locale}_${initiative}_${badgeType}_${source}.png`;
  img.crossOrigin = 'Anonymous';
  img.onload = function () {
    pdf.addImage(img, 50, 10, 105, 130);
    pdf.save(`${badgeType}-badge.pdf`);
  };

  pdf.text(`Badge Id : ${badgeId}`, 36, 190);
};

export const groupBy = function (list: any, keyName: any) {
  if (!list || list.length === 0 || !list[0][keyName]) {
    return [];
  }
  return list.reduce(function (pair: any, key: any) {
    (pair[key[keyName]] = pair[key[keyName]] || []).push(key);
    return pair;
  }, {});
};

/* istanbul ignore next */
export const visualize = (visualizer: any, analyser: any) => {
  const canvasCtx = visualizer.getContext('2d');
  const bufferLength = analyser.frequencyBinCount;
  const dataArray = new Uint8Array(bufferLength);
  const WIDTH = visualizer.width;
  const HEIGHT = visualizer.height;
  // TODO do we need to limit the number of time visualize refreshes per second
  // so that it can run on Android processors without causing audio to drop?
  function draw() {
    // this is more efficient than calling with processor.onaudioprocess
    // and sending floatarray with each call...
    requestAnimationFrame(draw);
    analyser.getByteTimeDomainData(dataArray);
    // canvasCtx.fillStyle = 'rgb(255, 255, 255, 0)';
    // canvasCtx.fillRect(0, 0, WIDTH, HEIGHT);
    canvasCtx.clearRect(0, 0, WIDTH, HEIGHT);
    canvasCtx.lineWidth = 2;
    canvasCtx.strokeStyle = 'rgb(0,123,255)';
    canvasCtx.beginPath();
    const sliceWidth = (WIDTH * 1.0) / bufferLength;
    let x = 0;
    for (let i = 0; i < bufferLength; i++) {
      let v = dataArray[i] / 128.0; // uint8
      let y = (v * HEIGHT) / 2; // uint8
      if (i === 0) {
        canvasCtx.moveTo(x, y);
      } else {
        canvasCtx.lineTo(x, y);
      }
      x += sliceWidth;
    }
    canvasCtx.lineTo(visualizer.width, visualizer.height / 2);
    canvasCtx.stroke();
  }

  draw();
};

export const getErrorMsg = (error: any) => {
  if (error && error.status === ErrorStatusCode.TOO_MANY_REQUEST) {
    return 'multipleRequestApiError';
  } else {
    return 'apiFailureError';
  }
};
