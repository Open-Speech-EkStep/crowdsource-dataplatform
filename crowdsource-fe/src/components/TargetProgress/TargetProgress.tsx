import { useEffect } from 'react';

import { Trans, useTranslation } from 'next-i18next';
import ProgressBar from 'react-bootstrap/ProgressBar';

import apiPaths from 'constants/apiPaths';
import { INITIATIVES_MAPPING } from 'constants/initiativeConstants';
import localStorageConstants from 'constants/localStorageConstants';
import useFetch from 'hooks/useFetch';
import useLocalStorage from 'hooks/useLocalStorage';
import type { CumulativeCount } from 'types/CumulativeCount';
import { convertTimeFormat, roundOffValue } from 'utils/utils';

import styles from './TargetProgress.module.scss';

const getSourceCount = (
  source: string | undefined,
  contributionCount: number | undefined,
  validationCount: number | undefined
) => {
  if (source === 'contribute') {
    return contributionCount ?? 0;
  } else if (source == 'validate') {
    return validationCount ?? 0;
  }
  return (contributionCount ?? 0) + (validationCount ?? 0);
};

const reduceList = (dataList: any) => {
  return dataList.reduce((accumulator: any, item: any) => {
    Object.keys(item).forEach(key => {
      if (typeof item[key] === 'string') return;
      accumulator[key] = (accumulator[key] || 0) + item[key];
    });
    return accumulator;
  }, {});
};

interface TargetProgressProps {
  initiative: string;
  initiativeType: string;
  source?: string;
  language?: string;
}

const TargetProgress = (props: TargetProgressProps) => {
  const { t } = useTranslation();

  const { data: cumulativeCountData, mutate: cumulativeMutate } = useFetch<Array<CumulativeCount>>(
    apiPaths.cumulativeCount,
    { revalidateOnMount: false }
  );

  const { data: initiativeGoalData, mutate: initiativeGoalMutate } = useFetch<
    Array<{ contribution_goal: number; type: string; validation_goal: number }>
  >(apiPaths.initiativeGoals, { revalidateOnMount: false });

  const [contributionLanguage] = useLocalStorage<string>(localStorageConstants.contributionLanguage);

  useEffect(() => {
    if (contributionLanguage) {
      cumulativeMutate();
      initiativeGoalMutate();
    }
  }, [initiativeGoalMutate, cumulativeMutate, contributionLanguage]);

  let totalProgress;
  let formattedAverage;
  let initiativeGoal;
  let currentInitiativeProgress;
  let initiativeUnit;

  const getInitiativeGoal = (data: any) => {
    const goalDataList = data.filter((d: any) => d.type === props.initiativeType) || [];
    const goalData = reduceList(goalDataList);
    const totalGoal = getSourceCount(props.source, goalData.contribution_goal, goalData.validation_goal) || 1;
    return totalGoal;
  };

  if (cumulativeCountData && cumulativeCountData.length && initiativeGoalData && initiativeGoalData.length) {
    const initiativeCumulativeData = cumulativeCountData.find(item => item.type === props.initiativeType);
    if (props.initiative === INITIATIVES_MAPPING.suno || props.initiative === INITIATIVES_MAPPING.bolo) {
      initiativeUnit = t('hours');
      totalProgress = getSourceCount(
        props.source,
        initiativeCumulativeData?.total_contributions,
        initiativeCumulativeData?.total_validations
      );

      currentInitiativeProgress = convertTimeFormat(totalProgress);
    } else {
      initiativeUnit = props.initiative === INITIATIVES_MAPPING.likho ? t('sentences') : t('images');
      totalProgress = getSourceCount(
        props.source,
        initiativeCumulativeData?.total_contribution_count,
        initiativeCumulativeData?.total_validation_count
      );
      currentInitiativeProgress = totalProgress;
    }
    initiativeGoal = getInitiativeGoal(initiativeGoalData);
    formattedAverage = roundOffValue((totalProgress / initiativeGoal) * 100, 1);
  }

  const initiativeName = `${t(props.initiative)} ${t('india')}`;

  if (!cumulativeCountData) return null;
  return (
    <div className={`${styles.root} d-flex flex-column`} data-testid="TargetProgress">
      <div className={`${styles.details} d-flex justify-content-between align-items-center`}>
        <div className={styles.percentage}>
          <Trans
            i18nKey="progressStatus"
            defaults="progressStatus"
            values={{ average: formattedAverage, initiativeName }}
            components={{ span: <span className={styles.count} /> }}
          />
        </div>
      </div>
      <ProgressBar
        now={formattedAverage}
        className={`${styles.progressBar} mt-2 mt-md-1 border border-1 border-info`}
      />
      <span className={`${styles.hours} mt-2 align-self-end`}>
        {currentInitiativeProgress}/{initiativeGoal} {initiativeUnit}
      </span>
    </div>
  );
};

export default TargetProgress;
