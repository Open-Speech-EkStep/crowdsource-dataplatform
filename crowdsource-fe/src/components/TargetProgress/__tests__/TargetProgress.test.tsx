import { when } from 'jest-when';

import { render, screen, waitFor } from 'utils/testUtils';

import TargetProgress from '../TargetProgress';

describe('TargetProgress', () => {
  const setup = async (
    initiative: any,
    initiativeMedia: string,
    source?: string,
    contributionLanguage?: string
  ) => {
    when(localStorage.getItem)
      .calledWith('contributionLanguage')
      .mockImplementation(() => 'Hindi');

    when(localStorage.getItem)
      .calledWith('likho_to-language')
      .mockImplementation(() => 'English');

    fetchMock.doMockOnceIf('/aggregated-json/cumulativeCount.json').mockResponseOnce(
      JSON.stringify([
        {
          total_contribution_count: 45,
          total_contributions: 0.031,
          total_languages: 2,
          total_speakers: 11,
          total_validation_count: 8,
          total_validations: 0.002,
          type: 'asr',
        },
        {
          total_contribution_count: 45,
          total_contributions: 0.0,
          total_languages: 3,
          total_speakers: 11,
          total_validation_count: 9,
          total_validations: 0.0,
          type: 'ocr',
        },
        {
          total_contribution_count: 45,
          total_contributions: 0.0,
          total_languages: 2,
          total_speakers: 10,
          total_validation_count: 8,
          total_validations: 0.0,
          type: 'parallel',
        },
        {
          total_contribution_count: 45,
          total_contributions: 6.3,
          total_languages: 2,
          total_speakers: 11,
          total_validation_count: 8,
          total_validations: 0.2,
          type: 'text',
        },
      ])
    );

    fetchMock.doMockOnceIf('/aggregated-json/initiativeGoals.json').mockResponseOnce(
      JSON.stringify([
        {
          contribution_goal: 60000,
          type: 'parallel',
          validation_goal: 60000,
        },
        {
          contribution_goal: 60000,
          type: 'ocr',
          validation_goal: 60000,
        },
        {
          contribution_goal: 60,
          type: 'asr',
          validation_goal: 60,
        },
        {
          contribution_goal: 60,
          type: 'text',
        },
      ])
    );

    const renderResult = render(
      <TargetProgress
        initiative={initiative}
        initiativeType={initiativeMedia}
        source={source}
        language={contributionLanguage}
      />
    );
    await waitFor(() => {
      expect(localStorage.getItem).toBeCalled();
    });
    await waitFor(() => {
      expect(fetchMock).toBeCalledWith('/aggregated-json/cumulativeCount.json');
    });
    await waitFor(() => {
      expect(fetchMock).toBeCalledWith('/aggregated-json/initiativeGoals.json');
    });
    return renderResult;
  };

  it('should render the component and matches it against stored snapshot', async () => {
    const { asFragment } = await setup('suno', 'asr');

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render the result for suno initiative home page', async () => {
    await setup('suno', 'asr', 'contribute');
    expect(screen.getByText('progressStatus')).toBeInTheDocument();
    expect(
      screen.getByText(content => {
        return content.includes('1 minutes/60 hours');
      })
    ).toBeInTheDocument();
  });

  it('should render the result for dekho initiative validate', async () => {
    await setup('dekho', 'ocr', 'validate');
    expect(screen.getByText('progressStatus')).toBeInTheDocument();
    expect(
      screen.getByText(content => {
        return content.includes('9/60000 images');
      })
    ).toBeInTheDocument();
  });

  it('should render the result for likho initiative contribute', async () => {
    await setup('likho', 'parallel', 'contribute');

    expect(screen.getByText('progressStatus')).toBeInTheDocument();
    expect(
      screen.getByText(content => {
        return content.includes('45/60000 sentences');
      })
    ).toBeInTheDocument();
  });

  it('should render the result for likho initiative', async () => {
    await setup('likho', 'parallel');

    expect(screen.getByText('progressStatus')).toBeInTheDocument();
    expect(
      screen.getByText(content => {
        return content.includes('53/120000 sentences');
      })
    ).toBeInTheDocument();
  });

  it('should print goal as 1 if goal is undefined', async () => {
    await setup('bolo', 'text', 'validate');

    expect(
      screen.getByText(content => {
        return content.includes('/1 hours');
      })
    ).toBeInTheDocument();
  });
});
