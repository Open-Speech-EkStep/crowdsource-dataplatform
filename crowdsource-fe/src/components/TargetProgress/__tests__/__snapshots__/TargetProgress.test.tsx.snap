// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TargetProgress should render the component and matches it against stored snapshot 1`] = `
<DocumentFragment>
  <div
    class="root d-flex flex-column"
    data-testid="TargetProgress"
  >
    <div
      class="details d-flex justify-content-between align-items-center"
    >
      <div
        class="percentage"
      >
        progressStatus
      </div>
    </div>
    <div
      class="mt-2 mt-md-1 progress"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0.0"
        class="progress-bar"
        role="progressbar"
        style="width: 0%;"
      />
    </div>
    <span
      class="hours mt-2 align-self-end"
    >
      1 minutes/120 hours
    </span>
  </div>
</DocumentFragment>
`;
